src/scripts/main.js

```diff
@@ -1 +1,63 @@
 'use strict';
+
+const successEl = (message) =>
+  `<div data-qa="notification" class="success">${message}</div>`;
+const errorEl = (message) =>
+  `<div data-qa="notification" class="error">${message}</div>`;
+
+const promiseHandler = (message) => {
+  const body = document.querySelector('body');
+
+  body.insertAdjacentHTML('beforeend', message);
+};
+
+const promise1 = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    reject(errorEl('First promise was rejected'));
+  }, 3000);
+
+  document.addEventListener('click', () => {
+    clearTimeout(timeoutId);
+    resolve(successEl('First promise was resolved'));
+  });
+});
```

Comment:
In this block of code, you are creating a new Promise that is either resolved or rejected based on a user action (click). However, you are not removing the event listener after the Promise has been settled. This could potentially lead to memory leaks if not handled properly. Consider removing the event listener once the Promise is settled.